email_backend/data_management/generate_csv.js

```diff
@@ -6,86 +6,106 @@ import fs from "fs";
 import mongoose from "mongoose";
 import { Parser } from "json2csv";
 
-const json2csv_parser = new Parser();
-
 const templates = get_templates();
-connect_db()
-  .then(() =>
-    _.chain(templates)
-      .map((template_value, template_name) => {
-        return make_mongoose_model_from_original_template({
-          original_template: template_value,
-          template_name,
-        });
-      })
-      .map((model) => model.find({}).exec())
-      .thru((promises) => Promise.all(promises))
-      .value()
-  )
-  .then((all_email_logs) => {
-    //if it happens that later on we add another key with a json value,
-    //change reduced json to handle "key.value". For the time being, the CSV
-    //will be cleaner if we flatten the "key.value" to "value"
-    _.forEach(all_email_logs, (collection, index) => {
-      if (collection.length > 0) {
-        const template_name = _.keys(templates)[index];
+const csv_strings = [];
+export const extractor = (testing = false) =>
```

Comment:
This flag doesn't seem to be in use in the test? 

Unless super necessary, it's much better not to use something like this though. You can't reach 100% test coverage if your code branches based on wether you're in test mode. In this case, rather than not calling `fs.writeFile` when testing, I'd suggest mocking `fs.writeFile` to a no-op (or better, mock it to something that validates the arguments passed in and throws if they are invalid).